internal/api/token_refresh.go

```diff
@@ -64,31 +64,18 @@ func (a *API) RefreshTokenGrant(ctx context.Context, w http.ResponseWriter, r *h
 		}
 
 		if session != nil {
-			var notAfter time.Time
+			result := session.CheckValidity(retryStart, &token.UpdatedAt, config.Sessions.Timebox, config.Sessions.InactivityTimeout)
```

Comment:
/nit Is it possible for a session to have both  exceeded `NotAfter`  and also  have`TimedOut`?

In such a scenario it'd return the more general default error (NotAfter) rather than the TimedOut message. If such a case doesn't exist I guess it's fine